lib/yi_appium_caps_util.rb

```diff
@@ -213,6 +219,57 @@ def update_mac_caps (output_data)
       raise "An error of type #{ex.class} happened, message is #{ex.message}"
     end
 
+    def update_tvos_caps (output_data)
+      #Replace value of deviceName
+      output_data['caps']['deviceName'] = "Apple TV"
+      output_data['parameters']['formFactor'] = "10foot"
+
+      if (output_data['caps']['skipYouiEngineAppAddress'] != nil)
+        puts
+        puts 'Skiping IP check since skipYouiEngineAppAddress is present in appium.txt'
+      else
+        puts 'Searching for IP, this may take a few seconds...'
+
+        # uswish/Test/tools/getIOSIP
+        myAppFolder = File.dirname(__FILE__) + "/../app/"
+        myApp = myAppFolder + "getIP.app"
+        myZipApp = myAppFolder + "getIP-tvOS.zip"
+        %x[unzip -o #{myZipApp} -d #{myAppFolder}]
+
+        puts "Launching getIP app"
+        #Launching app and putting (5 seconds of) log into file
+        ipcounter = 1
+        sleepSec = 7
+        while ipcounter <= 5 do
+          puts "Try \# #{ipcounter}"
+          puts "Sleep: #{sleepSec} sec"
+          logs = %x[ios-deploy --justlaunch --bundle #{myApp} & (idevicesyslog) & sleep #{sleepSec} ; kill $!]
+          File.write('logs.txt', logs)
+          #Getting ip from file
+          ip = %x[grep -Eo "IPAddress.*" logs.txt | head -1 | awk '{print $3}'| tr -d '"']
+          #Remove whitespace
+          ip = ip.strip
```

Comment:
Could you append `.strip` to the end of line 249 to avoid this step?

Reply:
didn't work